WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java

```diff
@@ -151,4 +153,47 @@ public void onCancel(DialogInterface dialog) {
         // Do not ask again
         AppPrefs.setImageOptimizePromoRequired(false);
     }
+
+    /**
+     * Given a media error returns the error message to display on the UI.
+     *
+     * @param error The media error occurred
+     * @return String  The associated error message.
+     */
+    public static String getErrorMessage(final Context context, final MediaModel media, final MediaStore.MediaError error) {
+        if (context == null || media == null || error == null) {
+            return null;
+        }
+
+        switch (error.type) {
+            case FS_READ_PERMISSION_DENIED:
+                return context.getString(R.string.error_media_insufficient_fs_permissions);
+            case NOT_FOUND:
+                return context.getString(R.string.error_media_not_found);
+            case AUTHORIZATION_REQUIRED:
+                return context.getString(R.string.media_error_no_permission_upload);
+            case REQUEST_TOO_LARGE:
+                if (media.isVideo()) {
+                    return context.getString(R.string.media_error_http_too_large_video_upload);
+                } else {
+                    return context.getString(R.string.media_error_http_too_large_photo_upload) + ". " +
+                            context.getString(R.string.media_error_suggest_optimize_image);
+                }
+            case SERVER_ERROR:
+                return context.getString(R.string.media_error_internal_server_error);
```

Comment:
We've seen that most of the times internal server errors are due to memory problem uploading pictures. The server may have other problems too. Should we remove the optimize image section of it?